keras/backend/tensorflow_backend.py

```diff
@@ -11,7 +11,7 @@
 _SESSION = None
 _LEARNING_PHASE = tf.placeholder(dtype='uint8', name='keras_learning_phase')  # 0 = test, 1 = train
 _MANUAL_VAR_INIT = False
-
+_DEVICE = None
```

Comment:
We definitely require a global device variable (at least for theano). This is because shared variables should be stored in the right device, else they will have to moved around GPU to GPU during runtime.  In tensorflow backend this could be eliminated, but I thought it would be better for both backends to have a similar implementation.